components-java-sdk/src/main/java/io/dapr/components/domain/state/StateStore.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.domain.state;
+
+import io.dapr.components.aspects.AdvertisesFeatures;
+import io.dapr.components.aspects.InitializableWithProperties;
+import io.dapr.components.aspects.Pingable;
+import reactor.core.publisher.Mono;
+
+
+public interface StateStore extends InitializableWithProperties, AdvertisesFeatures, Pingable {
```

Comment:
As default implementations of `AdvertisesFeatures` and `Pingable` can be provided at the gRPC wrapper layer, should these be optional interfaces?

Reply:
I can provide “default” definitions for those in their interfaces instead. It’s… Java version of “not supporting” multiple inheritance. 🤣

Reply:
Addressed in #16